### PR TITLE
LPS-128735 Translation in CKEditor source mode is lost upon switching locales

### PR DESCRIPTION
**Pull request by EliveltonR18**
Hi, reviewer! The PR is related with [this PTR](https://issues.liferay.com/browse/PTR-2340) and with [this discussion](https://github.com/liferay-data-engine/liferay-portal/pull/541) too.

The related issue [is here](https://issues.liferay.com/browse/LPS-128735).

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js [hunk 1]
```diff
@@ -51,6 +46,8 @@ const RichText = ({
 		}
 	}, [editingLanguageId, editorRef]);
 
+	const currentValue = predefinedValue || value;
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js:49]: - I believe `value` should came first, so the `predefinedValue` will be used when no value is provide. 
- Another point to attention is the use or the operator OR (`||`),  use the [Nullish coalescingoperator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator), so once the input value is changed `value` will be display in place of the `predefinedValue` (even when it is an empty string).

```suggestion
	const currentValue = value ?? predefinedValue;
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js [hunk 2]
```diff
@@ -62,32 +59,20 @@ const RichText = ({
...
-							setCurrentValue(dataValue);
-
-							onChange({}, dataValue);
-						});
-					}
-					else {
-						editor.removeListener('afterSetData');
+				onSetData={(event) => {
+					if (event.editor.mode === 'source') {
+						onChange({target: {value: event.data.dataValue}});
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js:75]: It is completely optional, but you can take advantage of the [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)
```suggestion
				onSetData={({data: {dataValue: value}, editor: {mode}}) => {
					if (mode === 'source') {
						onChange({target: {value}});
```

**Review by miltonmc — CHANGES_REQUESTED** [review]
Good work!

Just a few things I've noticed:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js (hunk 1 above)
Comment (EliveltonR18) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js:49]: Hi, @miltonmc I will do the changes!

**Review by miltonmc — APPROVED** [review]
LGTM!

**Comment by EliveltonR18** [comment]
ci:test:relevant

**Comment by EliveltonR18** [comment]
ci:test:sf

**Comment by EliveltonR18** [comment]
I executed the tests locally and the fail wasn't reproduced, i will run the tests again.

**Comment by EliveltonR18** [comment]
ci:reevaluate:1277401_69

**Comment by EliveltonR18** [comment]
ci:test:relevant

**Comment by EliveltonR18** [comment]
I will resend in other PR, the CI suite test wasn't called

**Comment by EliveltonR18** [comment]
ci:close

**Comment by miltonmc** [comment]
ci:stop

**Pull request closed without merging**